h2o-algos/src/main/java/hex/glrm/GLRMModel.java

```diff
@@ -228,19 +228,20 @@ private Frame reconstruct(Frame orig, Frame adaptedFr, Key<Frame> destination_ke
     Frame fullFrm = new Frame(adaptedFr);
     Frame loadingFrm = null;  // get this from DKV or generate it from scratch
     // call resconstruct only if test frame key and training frame key matches plus frame dimensions match as well
-    if ((_parms.train() == null) || orig.checksum()!=(_parms.train().checksum()) ||
-            !(orig._key.equals(_parms.train()._key))) { // compare with checksum instead of frame keys.
+    try { // try to use the old X frame if possible
+      loadingFrm = DKV.get(_output._x_factor_key).get();
+    } catch (Exception ex) {
```

Comment:
We should be able to detect this state better than by catching an exception. One way is to just check ` DKV.get(_output._x_factor_key) != null`.

Ideally, we should now the model was reloaded and that we do not have the frame anymore. But local fix would also be okay because detecting model reload is not a trivial problem.

Reply:
Yes, that is better.  Done.   Thanks,
W